test/mocks/config/mocks.h

```diff
@@ -26,14 +26,22 @@ template <class ResourceType> class MockSubscription : public Subscription<Resou
 
 class MockAdsWatch : public AdsWatch {
 public:
+  MockAdsWatch();
+  virtual ~MockAdsWatch();
```

Comment:
nit: virtual here and below is redundant as base class already has virtual destructor. (Typically we don't do this in code for brevity).

Reply:
Will clean this up in following PR.